packages/frontend/src/features/sqlRunner/store/history.ts

```diff
@@ -0,0 +1,40 @@
+import { type PayloadAction } from '@reduxjs/toolkit';
+
+export interface HistoryItem<T> {
+    value: T;
+    timestamp: number;
+}
+
+export interface WithHistory<T> {
+    current: T;
+    past: HistoryItem<NonNullable<T>>[];
+}
+
+const DEFAULT_MAX_HISTORY_ITEMS = 10;
+
+export const createHistoryReducer = <T>(
```

Comment:
Correct me if I'm wrong, but because history is just a redux component,a nd not persisted in database, this information will be lost if you refresh the page, and it is not shared across users. 

This is similar than doing CTRL-Z, right ? 

Are we considering persisting this info in database or even in session storage ?

Reply:
I've just tested this and if I refresh the page I lose all the history.

Reply:
> Are we considering persisting this info in database or even in session storage ?

Ideally it should be stored in the database when we start implementing draft state for this particular feature.

> This is similar than doing CTRL-Z, right ?

_Could_ be similar, but this only stores run queries, not code that is written, if that makes sense 👍 

> I've just tested this and if I refresh the page I lose all the history.

I'm ok with that, we also lose all config if we refresh the sql runner page when creating a new one IIRC